### DISCUSSION TITLE
Sorting Table by relation

### DISCUSSION THREAD

**Discussion by cleargoal** [filamentphp/filament#13449] [Help]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

A resource has the relational field
`TextColumn::make('lastContribution.amount')->money('USD', divideBy: 100)->width('5rem')->alignment(Alignment::Center)
                    ->sortable()->label('Розмір внеску'),` it gets data correctly but doesn't sort by this field.

Also, I want to set the default sorting by this field like this `->defaultSort('lastContribution.amount', 'desc')` but it doesn't sort either.

What is the correct way to sort by related field?

I am kindly grateful for any help.

**Comment by leandrocfe** [comment]
I think you can pass a closure

```php
->defaultSort(function ($query, $direction) {
    $query->orderBy(..);
})
```

**Reply by cleargoal** [reply]
Thank you!
